flink-libraries/flink-table/src/test/scala/org/apache/flink/table/utils/CommonTestData.scala

```diff
@@ -98,4 +106,134 @@ object CommonTestData {
       this(null, null)
     }
   }
+
+  def getMockTableEnvironment: TableEnvironment = new MockTableEnvironment
+
+  def getFilterableTableSource(
+    fieldNames: Array[String] = Array[String](
+      "name", "id", "amount", "price"),
+    fieldTypes: Array[TypeInformation[_]] = Array(
+      BasicTypeInfo.STRING_TYPE_INFO,
+      BasicTypeInfo.LONG_TYPE_INFO,
+      BasicTypeInfo.INT_TYPE_INFO,
+      BasicTypeInfo.DOUBLE_TYPE_INFO)) = new TestFilterableTableSource(fieldNames, fieldTypes)
+}
+
+class MockTableEnvironment extends TableEnvironment(new TableConfig) {
+
+  override private[flink] def writeToSink[T](table: Table, sink: TableSink[T]): Unit = ???
+
+  override protected def checkValidTableName(name: String): Unit = ???
+
+  override protected def getBuiltInRuleSet: RuleSet = ???
+
+  override def sql(query: String): Table = ???
+
+  override def registerTableSource(name: String, tableSource: TableSource[_]): Unit = ???
+}
+
+class TestFilterableTableSource(
+    fieldNames: Array[String],
```

Comment:
I think it is OK to hard-code the field names and types to make the test class more concise.